data-serving/data-service/src/controllers/case.ts

```diff
@@ -29,28 +29,48 @@ export const get = async (req: Request, res: Response): Promise<void> => {
  * Handles HTTP GET /api/cases/download.
  */
 export const download = async (req: Request, res: Response): Promise<void> => {
-    const cases = await Case.find({}).lean();
+    let cases: any;
+    try {
+        if (req.query.q) {
+            cases = await (casesMatchingSearchQuery({
+                searchQuery: req.query.q as string,
+                count: false,
+            }) as DocumentQuery<CaseDocument[], CaseDocument, unknown>).lean();
```

Comment:
wondering if we want to move the lean() inside caseMatchingSearchQuery so that callers don't have to care about it?

Reply:
Done.